docs/app/src/versions.js

```diff
@@ -2,36 +2,70 @@
 
 angular.module('versions', [])
 
-.controller('DocsVersionsCtrl', ['$scope', '$location', '$window', 'NG_VERSIONS', function($scope, $location, $window, NG_VERSIONS) {
+.controller('DocsVersionsCtrl', [
+  '$scope',
+  '$location',
+  '$window',
+  '$http',
+  '$q',
+  'NG_VERSIONS', function($scope, $location, $window, $http, $q, NG_VERSIONS) {
+
   $scope.docs_version  = NG_VERSIONS[0];
   $scope.docs_versions = NG_VERSIONS;
 
-  for (var i = 0, minor = NaN; i < NG_VERSIONS.length; i++) {
-    var version = NG_VERSIONS[i];
-    if (version.isSnapshot) {
-      version.isLatest = true;
-      continue;
-    }
-    // NaN will give false here
-    if (minor <= version.minor) {
-      continue;
+  // If this is not the snapshot version, request the snapshot's versions data
+  // to fill the version list with current data
+  $q(function(resolve, reject) {
+    if ($scope.docs_version.isSnapshot) {
+      reject();
+    } else {
+      resolve();
     }
-    version.isLatest = true;
-    minor = version.minor;
-  }
+  }).then(function() {
+    return $http.get('../snapshot/versions_data.json');
```

Comment:
Yes. It will also fail silently if the snapshot json data can't be requested for some reason. It's strictly an optional enhancement.